Disk Alias configuration during deploy VM from Template

pavel-z1 opened the pull request:
Add possibility to setup Disk Alias during deploy VM from Template

Example:
```
resource "ovirt_vm" "vm_name" {
...
 
    block_device {
        interface      = "virtio_scsi"
        alias          = "vm_name_Disk1"
    }

    initialization {
        ...
}
```

pavel-z1 commented:
Was added additionally option storage_domain.
storage_domain define destination storage domain to which will be copied disks during deploy VM from Template.

Based on example from go_ovirt - https://github.com/oVirt/go-ovirt/blob/90013aa942c314a9bfa453c1907ba2ad4445bccf/examples/add_vm_from_template.go

pavel-z1 commented:
Hi @rgolangh 
Can you please approve PR?

rgolangh commented:
this PR contains more changes than the alias addition. Please leave only the related stuff for alias.

pavel-z1 commented:
Hi @rgolangh , this Pr was created more than month ago.
In my production environment I have already dependencies from this PR.

Unfortunately, I don't have enough free time to back again, split commits t the different PRs and perform again full own test to ensure that will not occurred issues.

This PR doesn't have commits  which relate to different components.
Commits from this PR related to the disk parameters configuration during deploy VM from Template.

I ask you accept this PR as it is. To prevent wasting time for splitting commits.

I've changed PR alias so that it displays content more correctly.

rgolangh commented:
Your contribution is very valuable and have direct affect, which you probably don't know on other users and projects, which rely on it. This means that every commit, every change, is important. 
Splitting the work to logical commits, and making a logical change in a PR is a must, and for the sake of the whole project and its users I'm not going to ignore that.
Really shouldn't be a lot of work because you already made all the coding, just some git practice and you have it.

pavel-z1 commented:
Hi @rgolangh,

I think such questions would not have arisen if Pull Requests had not been left without consideration for a long time.
Regardless of the complexity of PR, it is difficult to return to the code after a month and a half.

In any case, I've divided PR to the two:
https://github.com/oVirt/terraform-provider-ovirt/pull/214
https://github.com/oVirt/terraform-provider-ovirt/pull/216

Can you please review them?

rgolangh commented on ovirt/resource_ovirt_vm.go:
```diff
@@ -938,9 +943,16 @@ func expandOvirtVMDiskAttachment(d interface{}, disk *ovirtsdk4.Disk) (*ovirtsdk
 		if v, ok := dmap["size"]; ok {
-			newSize := int64(v.(int)) * int64(math.Pow(2, 30))
-			disk.SetProvisionedSize(newSize)
+			if v != 0 {
```
Please remove the change for the provisionedSize. If there's no value the `ok` variable will be false.

pavel-z1 replied:
The condition `if v != 0 {` is necessary for this scenario:
```
resource "ovirt_vm" "vm" {
    ...
    template_id          = data.ovirt_templates.search_filtered_template.templates[0].id
    ...
    block_device {
        interface = var.vm_disk_interface
        alias = "vmname_Boot"
    }
}
```

When I need create new VM from template, I want to change only disk alias or interface type.
I don't need provision the new block_device.
By default, disk size return 0 if size parameter is not defined.

Therefore this check is necessary.
Without this check terraform will try setup disk size 0 for disk from template. Such attempts cause errors.

rgolangh commented on ovirt/resource_ovirt_vm.go:
```diff
@@ -938,9 +943,16 @@ func expandOvirtVMDiskAttachment(d interface{}, disk *ovirtsdk4.Disk) (*ovirtsdk
+	        if v, ok := dmap["alias"]; ok {
+			if v != "" {
+	                	disk.SetAlias(v.(string))
+			}
```
Please squash the next commit for formatting with this one.

pavel-z1 replied:
Done

rgolangh commented on ovirt/resource_ovirt_vm.go:
```diff
@@ -174,6 +174,11 @@ func resourceOvirtVM() *schema.Resource {
+                                                        Type:     schema.TypeString,
+                                                        Optional: true,
+                                                        ForceNew: false,
+                                                },
```
Please squash the next formatting commit with this one.

pavel-z1 replied:
Done

pavel-z1 commented:
Hi @rgolangh  , could we are finish with pull request?

pavel-z1 commented:
Hi @rgolangh , please approve this Pull Request.

pavel-z1 commented:
Hi @imjoey!
Seems Roy currently busy. Can you help with review and approval of this PR?

emesika commented:
Please resolve conflicts

pavel-z1 commented:
Hi @emesika 
PR rebased on master. Conflict resolved. Please merge PR.
